src/convertToObject.js

```diff
@@ -5,8 +5,38 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = {};
+
+  if (!sourceString) {
+    return stylesObject;
+  }
+
+  const stringWithoutComments = sourceString.replace(
+    /\/\*[^*]*\*+([^/*][^*]*\*+)*\//g,
+    '',
+  );
+
+  const matches = stringWithoutComments.match(/(?:[^;"']|"[^"]*"|'[^']*')+/g);
+  const cssDeclarations = matches
+    ? matches.map((decl) => decl.trim()).filter((decl) => decl)
+    : [];
```

Comment:
You don't need regular expressions for this task. Just split the string `;` remove spaces, and empty lines.